### PR TITLE
Finish TODO task for POW module

### PR DESCRIPTION
**Pull request by ShengguangXiao**
## Description

Finish TODO task for POW module

"TODO: Verify the PoW submission for this unknown node", implement the part to verify the PoW from DS leader

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libDirectoryService/DSBlockPreProcessing.cpp [hunk 1]
```diff
@@ -311,15 +309,11 @@ bool DirectoryService::VerifyPoWWinner(
                       << ". Will continue look for it in PoW from leader.");
       if (dsWinnerPoWsFromLeader.find(DSPowWinner.first) !=
           dsWinnerPoWsFromLeader.end()) {
-        uint8_t expectedDSDiff = DS_POW_DIFFICULTY;
+        uint8_t expectedDSDiff = m_mediator.m_dsBlockChain.GetLastBlock()
```
Comment (ansnunez) [inline src/libDirectoryService/DSBlockPreProcessing.cpp]: This change doesn't seem related to the to-do.  Was `expectedDSDiff = DS_POW_DIFFICULTY` a bug?

Comment (ShengguangXiao) [reply src/libDirectoryService/DSBlockPreProcessing.cpp]: No, previous the genesis block difficulty is not using DS_POW_DIFFICULTY, now it is fixed. So it is kind of code optimization

File: src/libDirectoryService/DSBlockPreProcessing.cpp [hunk 2]
```diff
@@ -436,11 +430,28 @@ bool DirectoryService::VerifyPoWOrdering(
...
+                  m_pendingDSBlock->GetHeader().GetBlockNum(), difficulty,
+                  headerHash, powSoln.nonce,
+                  DataConversion::charArrToHexStr(powSoln.result),
+                  DataConversion::charArrToHexStr(powSoln.mixhash))) {
+            LOG_GENERAL(WARNING,
+                        "Failed to verify PoW solution from leader for node: "
+                            << toFind);
+            ret = false;
+          }
+          result = powSoln.result;
```
Comment (sandipbhoir) [inline src/libDirectoryService/DSBlockPreProcessing.cpp]: Add this to else statement

**Comment by ShengguangXiao** [comment]
I specially modify the code, DS backup doesn't process PoW, but later verify the PoWs from the leader, the VerifyPoWOrdering can pass.
